HW8(MVC)/scripts/Controller.js

```diff
@@ -0,0 +1,138 @@
+import { $V } from './View.js';
+import { $M } from './Model.js';
+
+class Controller {
+  constructor () {
+    this.invalidInputTimer = null;
+    this.tasksListIsHidden = true;
+  }
+
+  initialize() {
+    $V.tasksContainer.addEventListener('click', this.handleClick);
+    $V.showButton.addEventListener('click', $V.toggleShowButton);
+    $V.addButton.addEventListener('click', this.addTask);
+
+    $V.input.addEventListener('input', () => {
+      if ($V.input.value.length > 0) {
+        $V.addButton.style.backgroundColor = '#87ff93';
+      } else {
+        $V.addButton.style.backgroundColor = '#fff';
+      }
+    });
+
+    if (this.anyTasksAlreadySaved()) {
+      $V.toggleShowButton();
+      $V.renderTasks($M.getTasksList());
+    }
+  }
+
+  handleClick(e) {
+    const clickedOn = e.target;
+    const clickedOnClassList = clickedOn.classList;
+  
+    if (clickedOnClassList.contains('icon')) {
+      let [type, index] = clickedOnClassList[1].split('-');
+
+      if (type === 'doneIcon') {
+        $C.toggleDoneMarker(index);
+      } else if (type === 'deleteIcon'){
+        $C.deleteTask(index);
+      } else {
+        $C.switchFilterMode();
+      }
+    }
+  }
+
+  toggleDoneMarker(index) {
+    const tasks = $M.getTasksList();
+    let isDone = tasks[index].isCompleted;
+
+    if (isDone) {
+      isDone = false;
+    } else {
+      isDone = true;
+    }
+    tasks[index].isCompleted = isDone;
+    
+    $M.setTasksList(tasks);
+    $V.renderTasks(tasks);
+  }
+
+  deleteTask(index) {
+    const tasks = $M.getTasksList();
+    tasks.splice(index, 1);
+
+    $M.setTasksList(tasks);
+    $V.renderTasks(tasks);
+  }
+
+  switchFilterMode() {
+    let currentStatus = this.isFiltered();
+
+    if (currentStatus === 'true') {
+      currentStatus = 'false';
+    } else {
+      currentStatus = 'true';
+    }
+
+    $M.setFilter(currentStatus);
```

Comment:
I`m using 'true' and 'false' as a string because localStorage saves only strings.

But I can save an object in localStorage, and then after JSON.parse() it will be boolean.

If that`s important, I can rewrite this logic. But as you mentioned, that`s just a suggestion. I will prefer not to, if possible.